apps/aecore/test/aec_conductor_tests.erl

```diff
@@ -381,6 +381,72 @@ test_block_publishing() ->
     ok.
 
 %%%===================================================================
+%%% Pending block tests
+%%%===================================================================
+
+block_candidate_test_() ->
+    {foreach,
+     fun() ->
+          TmpKeysDir = setup_common(),
+          {ok, _} = ?TEST_MODULE:start_link([{autostart, false}]),
+          meck:new(aec_mining, [passthrough]),
+          meck:expect(aec_mining, create_block_candidate,
+              fun(TopBlock, TopBlockState, AdjChain) ->
+                  timer:sleep(100),
+                  meck:passthrough([TopBlock, TopBlockState, AdjChain])
+              end),
+          meck:expect(aec_mining, mine,
+              fun(_, _) ->
+                  timer:sleep(3000),
+                  {error, no_solution}
+              end),
+          TmpKeysDir
+     end,
+     fun(TmpKeysDir) ->
+          meck:unload(aec_mining),
+          teardown_common(TmpKeysDir),
+          ok
+     end,
+     [
+      {"Get block candidate", fun test_get_block_candidate/0}
+     ]}.
+
+test_get_block_candidate() ->
+    assert_stopped_and_genesis_at_top(),
+    ?assertEqual({error, not_mining}, ?TEST_MODULE:get_block_candidate()),
+    {ok, MyAccount} = aec_keys:pubkey(),
+    lists:foreach(
+        fun(_) ->
+            Tx = #spend_tx{sender = MyAccount,
+                          recipient = MyAccount,
+                          nonce = 0, fee = 0},
+            {ok, STx} = aec_keys:sign(Tx),
+            ok = aec_tx_pool:push(STx, tx_received)
+        end,
+        lists:seq(1, 3)),
+    ?TEST_MODULE:start_mining(),
+    ?assertEqual({error, miner_starting}, ?TEST_MODULE:get_block_candidate()),
+    timer:sleep(2000),
```

Comment:
Is it possible to replace this with aec_test_utils:wait_for_it ?

Reply:
aec_test_utils:wait_for_it/2,3 needs an exact value to match in order to consider the condition met. Since we have no guarantee for the order of the transactions in the block, we can not use it.
I could implement a new function aec_test_utils:wait_for_it_not/2,3 that waits for the result NOT to be the one passed as a parameter (in our case - that the get_block_candidate/0 returns something different thant {error, miner_starting}